latch_cli/centromere/ctx.py

```diff
@@ -223,28 +238,98 @@ def full_image(self):
         """
         return f"{self.dkr_repo}/{self.image}"
 
-    def nucleus_provision_url(self) -> Tuple[str, str]:
+    def provision_register_deployment(self) -> Tuple[str, str]:
         """Retrieve centromere IP + username."""
+        print("Provisioning register instance. This may take a few minutes.")
 
-        headers = {"Authorization": f"Bearer {self.token}"}
+        resp = tinyrequests.post(
+            "https://centromere.latch.bio/register/start",
+            headers={"Authorization": f"Latch-SDK-Token {self.token}"},
+            json={"SSHKey": self.ssh_key_path.with_suffix(".pub").read_text()},
+        )
 
-        response = tinyrequests.post(
-            self.latch_provision_url,
-            headers=headers,
-            json={
-                "public_key": self.public_key,
-            },
+        json_data = resp.json()
+        if resp.status_code != 200:
+            raise ValueError(json_data["Error"])
+
+        ip = json_data["IP"]
+        self.jump_key_path.write_text(json_data["JumpKey"])
+        self.jump_key_path.chmod(0o600)
+
+        subprocess.run(
+            ["ssh-add", str(self.jump_key_path)],
+            stdout=subprocess.DEVNULL,
+            stderr=subprocess.DEVNULL,
         )
 
-        resp = response.json()
-        try:
-            public_ip = resp["ip"]
-            username = resp["username"]
-        except KeyError as e:
+        retries = 0
+        while retries < 120:
+            resp = tinyrequests.post(
+                "https://centromere.latch.bio/register/ready",
+                headers={"Authorization": f"Latch-SDK-Token {self.token}"},
+            )
+
+            if resp.status_code != 200:
+                raise ValueError(resp.json()["Error"])
+
+            if resp.json()["Ready"]:
+                break
+
+            retries += 1
+            time.sleep(1)
+
+        if retries == 120 or ip == "":
             raise ValueError(
-                f"Malformed response from request for access token {resp}"
-            ) from e
-        return public_ip, username
+                "Unable to provision registration server due to load - please try again"
+                " later."
+            )
+
+        return ip, "root"
+
+    def downscale_register_deployment(self):
+        resp = tinyrequests.post(
+            "https://centromere.latch.bio/register/stop",
+            headers={"Authorization": f"Latch-SDK-Token {self.token}"},
+        )
+
+        if resp.status_code != 200:
+            raise ValueError("unable to downscale register deployment")
+
+    def configure_ssh_config(
+        self,
+        internal_ip: str,
+    ):
+        self.ssh_config_path = Path.home() / ".ssh" / "config"
+
+        self.ssh_config_path.parent.mkdir(exist_ok=True)
+        self.ssh_config_path.touch(exist_ok=True)
```

Comment:
don't touch the file, handle reading errors instead

also if you create the file you need to set chmod i think